### PR TITLE
keep path suffix "/" if exists

### PR DESCRIPTION
**Pull request by jbking**
Path pattern can have suffix "/", like `/pets/` . In this case, the suffix "/" have to keep for requesting.
However its dropped by `path.Join()`

```
package main

import (
	"fmt"
	"path"
)

func main() {
	queryPath := "/prefix"
	fmt.Println(path.Join(queryPath, fmt.Sprintf("/pets")))
	fmt.Println(path.Join(queryPath, fmt.Sprintf("/pets/"))) // same output above
	fmt.Println(path.Join(queryPath, fmt.Sprintf("/pets/")) + "/") // keep suffix "/"
}
```

```
$ go run main.go
/prefix/pets
/prefix/pets
/prefix/pets/
```

Dropping the suffix "/" can cause a redirect response from a service for appending suffix "/". And it makes creating resource (POST) failed.

### CODE REVIEW

File: pkg/codegen/templates/client.tmpl [hunk 1]
```diff
@@ -165,6 +165,9 @@ func New{{$opid}}Request{{if .HasBody}}WithBody{{end}}(server string{{genParamAr
         return nil, err
     }
     queryUrl.Path = path.Join(queryUrl.Path, fmt.Sprintf("{{genParamFmtString .Path}}"{{range $paramIdx, $param := .PathParams}}, pathParam{{$paramIdx}}{{end}}))
+    if strings.HasSuffix("{{genParamFmtString .Path}}", "/") && ! strings.HasSuffix(queryUrl.Path, "/") {
```
Comment (deepmap-marcinr) [inline pkg/codegen/templates/client.tmpl]: Thank you for doing this, but this is a bit hacky and does a bunch of unnecessary string compares.

You already have queryUrl in the context, we can use that to concatenate paths properly while preserving trailing slashes - I should have thought of it when writing the code in the first place.

Try queryUrl.Parse(fmt.Sprintf("{{genParamFmtString .Path}}"{{range $paramIdx, $param := .PathParams}}, pathParam{{$paramIdx}}{{end}})

Basically, queryUrl.Parse can parse a relative url in the context of the server string.

**Comment by deepmap-marcinr** [comment]
I've committed an alternate fix myself.

File: pkg/codegen/templates/client.tmpl (hunk 1 above)
Comment (jbking) [reply pkg/codegen/templates/client.tmpl]: Thx!
